tasks/main.yml

```diff
@@ -40,14 +62,46 @@
     - { name: "docker-py", version: "1.10.5" }
     - { name: "docker-compose", version: "1.9.0" }
 
-- name: start docker
-  service: name=docker state=started enabled=yes
+- name: format docker storage
+  filesystem:
+    dev: "{{ docker_storage_device }}"
+    fstype: xfs
+  when: docker_storage_device is defined
   become: True
 
-- name: load overlay kernel module for docker
-  modprobe:
-    name: overlay
-    state: present
+- name: remove default mount
```

Comment:
This task seems to make assumptions about the operating systems that we probably don't want to make here. Namely that /mnt is mounted and /dev/vdb is a thing. We are already making an assumption that the operating system is RHEL/CentOS/something with both yum and SELinux. I think the assumptions made here are ones we could avoid making. Does the /mnt mount interfere with the setup in some way? If so, could the issues be avoided in some other way?

Reply:
True, this need rises directly from cloud-init setting /mnt/ mount on /dev/vdb by default on cPouta.